### PR TITLE
FW-4648 Add PATCH support to Site content APIs

### PR DESCRIPTION
**Pull request by NolanVH**
### Description of Changes
This PR enables PATCH for the following site content APIs:
- Categories at `/api/1.0/sites/{site_slug}/categories/{id}/`
- Characters at `/api/1.0/sites/{site_slug}/characters/{id}/`
- Dictionary at `/api/1.0/sites/{site_slug}/dictionary/{id}/`
- People at `/api/1.0/sites/{site_slug}/people/{id}/`
- Songs at `/api/1.0/sites/{site_slug}/songs/{id}/`
- Stories at `/api/1.0/sites/{site_slug}/stories/{id}/`
- Story Pages at `/api/1.0/sites/{site_slug}/stories/{story_pk}/pages/{id}/`
- Widgets at `/api/1.0/sites/{site_slug}/widgets/{id}/`

### Checklist
- [x] README / documentation has been updated if needed
- [x] PR title / commit messages contain Jira ticket number if applicable
- [x] Tests have been updated / created if applicable
- [x] Admin Panel has been updated if models have been added or modified
- [x] Migrations have been updated and committed if applicable
- [x] Team Postman workspace has been updated if applicable

### Reviewers Should Do The Following:
- [x] Review the code changes here
- [ ] Pull the branch and test locally

### Additional Notes
For reviewers, it would be good to test at least one of the PATCH API locally.

### CODE REVIEW

File: firstvoices/backend/tests/test_apis/base_api_test.py [hunk 1]
```diff
@@ -735,7 +735,7 @@ def test_destroy_site_missing_404(self):
         assert response.status_code == 404
 
 
-class SiteContentPatchApiTestMixin:
+class SiteContentPatchApiTestMixin(WriteApiTestMixin):
```
Comment (sarahfirstvoices) [inline firstvoices/backend/tests/test_apis/base_api_test.py:738]: I think it's best to not make this inherit from WriteApiTestMixin and instead just include a comment that they need to be used together. Having WriteApiTestMixin in the inheritance tree twice could cause surprising results in some tests, and it is already used in the base test classes

File: firstvoices/backend/views/dictionary_views.py [hunk 2]
```diff
@@ -92,7 +105,7 @@ class DictionaryViewSet(
     Dictionary entry information.
     """
 
-    http_method_names = ["get", "post", "put", "delete"]
+    http_method_names = ["get", "post", "put", "delete", "patch"]
```
Comment (sarahfirstvoices) [inline firstvoices/backend/views/dictionary_views.py:108]: Probably better to remove this line and keep the default of allowing all methods (including HEAD etc)

**Review by sarahfirstvoices — APPROVED** [review]
No need to re-review if you make the small changes I suggested. Looks good!

File: firstvoices/backend/tests/test_apis/test_category_api.py [hunk 3]
```diff
@@ -16,11 +17,16 @@
     get_app_admin,
     get_non_member_user,
 )
-from backend.tests.test_apis.base_api_test import BaseUncontrolledSiteContentApiTest
+from backend.tests.test_apis.base_api_test import (
+    BaseUncontrolledSiteContentApiTest,
+    SiteContentPatchApiTestMixin,
+)
```
[the remark this answers is not in the document]
Comment (NolanVH) [reply firstvoices/backend/tests/test_apis/test_category_api.py]: Good call, thanks. I have added the `SiteContentPatchApiTestMixin` to `BaseUncontrolledSiteContentApiTest`.

**Pull request merged**